src/sfloader/fluid_defsfont.c

```diff
@@ -1269,6 +1450,383 @@ fluid_zone_is_mod_identic(fluid_mod_t *mod, char *name)
     return FALSE;
 }
 
+/*
+ * Returns true if the modulator mod is in the path table, FALSE otherwise
+ * @param path path table of modulators.
+ * @count number of modulators in path.
+ * @mod modulator.
+ * @return TRUE if mod is in path table, FALSE otherwise.
+ */
+static int
+fluid_is_mod_in_path(fluid_mod_t *path[], int count, fluid_mod_t *mod)
+{
+    int i;
+    for (i = 0; i < count; i++)
+    {
+        if (path[i] == mod)
```

Comment:
`path[i]` will cause a buffer overflow when executing `test_modulator_links`. This probably has to do with the fact that the `path` pointer will be advanced when called from `fluid_defsfont.c:1556`, but `count` is no adjusted to reflect the remaining length of the buffer. Haven't been able to fully comprehend it yet, so I'm posting it here, so we don't forget about it. Here is the output of AddressSanitizer:

```
==11474==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x603000000360 at pc 0x00000044c22d bp 0x7fffffffd6f0 sp 0x7fffffffd6e8
READ of size 8 at 0x603000000360 thread T0
    #0 0x44c22c in fluid_is_mod_in_path fluid_defsfont.c:1466
    #1 0x44c712 in fluid_check_linked_mod_path fluid_defsfont.c:1556
    #2 0x44c52b in fluid_check_linked_mod_path fluid_defsfont.c:1524
    #3 0x44cd7f in fluid_zone_check_linked_mod fluid_defsfont.c:1641
    #4 0x40f103 in main test/test_modulator_links.c:145
    #5 0x7ffff3a2af89 in __libc_start_main (/lib64/libc.so.6+0x20f89)
    #6 0x40d229 in _start (build/test/test_modulator_links+0x40d229)

0x603000000360 is located 0 bytes to the right of 32-byte region [0x603000000340,0x603000000360)
allocated by thread T0 here:
    #0 0x7ffff6f00510 in malloc (/usr/lib64/libasan.so.4+0xdc510)
    #1 0x44ccef in fluid_zone_check_linked_mod fluid_defsfont.c:1634
    #2 0x40f103 in main test/test_modulator_links.c:145
    #3 0x7ffff3a2af89 in __libc_start_main (/lib64/libc.so.6+0x20f89)

SUMMARY: AddressSanitizer: heap-buffer-overflow fluid_defsfont.c:1466 in fluid_is_mod_in_path
Shadow bytes around the buggy address:
  0x0c067fff8010: fd fd fa fa fd fd fd fd fa fa fd fd fd fa fa fa
  0x0c067fff8020: fd fd fd fd fa fa 00 00 00 00 fa fa 00 00 00 00
  0x0c067fff8030: fa fa 00 00 00 00 fa fa 00 00 00 00 fa fa 00 00
  0x0c067fff8040: 00 00 fa fa 00 00 00 00 fa fa 00 00 00 00 fa fa
  0x0c067fff8050: fd fd fd fa fa fa fd fd fd fa fa fa fd fd fd fa
=>0x0c067fff8060: fa fa fd fd fd fd fa fa 00 00 00 00[fa]fa fa fa
  0x0c067fff8070: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c067fff8080: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c067fff8090: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c067fff80a0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c067fff80b0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==11474==ABORTING
```